### PR TITLE
Clear peeking ilgen callNode from callsite

### PR DESCRIPTION
**Pull request by cathyzhyi**
The callNode generated from peeking ilgen is used to create the
callsite of appropriate type and check for targets. Set it to NULL when
both are done so that inliner code can reliably figure out whether the
callsite has been updated with proper ilgen callNode or not.

Signed-off-by: Yi Zhang <yizhang@ca.ibm.com>

### CODE REVIEW

**Comment by cathyzhyi** [comment]
@andrewcraik Hey Andrew, can you review this PR? It's needed for fixing eclipse/openj9#7684

File: runtime/compiler/optimizer/J9EstimateCodeSize.cpp [hunk 1]
```diff
@@ -1239,9 +1239,10 @@ TR_J9EstimateCodeSize::realEstimateCodeSize(TR_CallTarget *calltarget, TR_CallSt
             TR::Node* parent = tt->getNode();
             TR::Node* callNode = tt->getNode()->getFirstChild();
             TR::SymbolReference* symRef =  callNode->getSymbolReference();
-            if (!callNode->getSymbolReference()->isUnresolved() && !visited.contains(callNode))
+            int i = callNode->getByteCodeIndex();
```
Comment (andrewcraik) [inline runtime/compiler/optimizer/J9EstimateCodeSize.cpp:1242]: I can see why you moved this out, but I worry that with such a common/simple name that is often used in loops we may get some weird interactions - could we just leave the `int i` inside and just add the extra call to `callNode->getByteCodeIndex()`. If we did want to pull it up we would need a better name IMO, but I don't think we want to consider a rename right now - just keep the scope as it was would be safest.

Comment (cathyzhyi) [reply runtime/compiler/optimizer/J9EstimateCodeSize.cpp:1242]: changed as suggested.

**Comment by cathyzhyi** [comment]
> One code issue. Other issue I wanted to check - are you sure we are NULL initializing that field properly already? eg we don't leave it uninitialized at present?

yes, we won't leave it uninitialized.

**Comment by andrewcraik** [comment]
Jenkins test sanity all jdk14

**Pull request merged**